diff --git a/node_modules/html-encoding-sniffer/lib/html-encoding-sniffer.js b/node_modules/html-encoding-sniffer/lib/html-encoding-sniffer.js
index 08a92d3..4e286e2 100644
--- a/node_modules/html-encoding-sniffer/lib/html-encoding-sniffer.js
+++ b/node_modules/html-encoding-sniffer/lib/html-encoding-sniffer.js
@@ -1,20 +1,21 @@
 "use strict";
-const { getBOMEncoding, labelToName } = require("@exodus/bytes/encoding-lite.js");
+let encodingLitePromise = import("@exodus/bytes/encoding-lite.js");
 
 // https://html.spec.whatwg.org/#encoding-sniffing-algorithm
-module.exports = (uint8Array, { xml = false, transportLayerEncodingLabel, defaultEncoding } = {}) => {
+module.exports = async (uint8Array, { xml = false, transportLayerEncodingLabel, defaultEncoding } = {}) => {
+  const encodingLite = await encodingLitePromise;
   if (defaultEncoding === undefined) {
     defaultEncoding = xml ? "UTF-8" : "windows-1252";
   }
 
-  let encoding = labelToName(getBOMEncoding(uint8Array));
+  let encoding = encodingLite.labelToName(encodingLite.getBOMEncoding(uint8Array));
 
   if (encoding === null && transportLayerEncodingLabel !== undefined) {
-    encoding = labelToName(transportLayerEncodingLabel);
+    encoding = encodingLite.labelToName(transportLayerEncodingLabel);
   }
 
   if (encoding === null && !xml) {
-    encoding = prescanMetaCharset(uint8Array);
+    encoding = await prescanMetaCharset(uint8Array, encodingLite);
   }
 
   if (encoding === null) {
@@ -25,7 +26,7 @@ module.exports = (uint8Array, { xml = false, transportLayerEncodingLabel, defaul
 };
 
 // https://html.spec.whatwg.org/multipage/syntax.html#prescan-a-byte-stream-to-determine-its-encoding
-function prescanMetaCharset(uint8Array) {
+async function prescanMetaCharset(uint8Array, encodingLite) {
   const l = Math.min(uint8Array.byteLength, 1024);
   for (let i = 0; i < l; i++) {
     let c = uint8Array[i];
